servicetalk-traffic-resilience-http/src/main/java/io/servicetalk/traffic/resilience/http/AbstractTrafficResilienceHttpFilter.java

```diff
@@ -377,4 +403,73 @@ public String toString() {
                     '}';
         }
     }
+
+    private static final class DebugTrackingDelegatingTicket implements Ticket {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(DebugTrackingDelegatingTicket.class);
+
+        private final Ticket delegate;
+        private volatile boolean completed;
+        @Nullable
+        private volatile Object terminationCause;
+
+        DebugTrackingDelegatingTicket(final Ticket delegate) {
+            this.delegate = delegate;
+        }
+
+        @Override
+        public CapacityLimiter.LimiterState state() {
+            completed = true;
+            return delegate.state();
+        }
+
+        @Override
+        public int completed() {
+            completed = true;
+            return delegate.completed();
+        }
+
+        @Override
+        public int dropped() {
+            completed = true;
+            return delegate.dropped();
+        }
+
+        @Override
+        public int failed(Throwable error) {
+            completed = true;
+            return delegate.failed(error);
+        }
+
+        @Override
+        public int ignored() {
+            completed = true;
+            return delegate.ignored();
+        }
+
+        @Override
+        public String toString() {
+            return "DebugTrackingDelegatingTicket{" +
+                    "delegate=" + delegate +
+                    ", completed=" + completed +
+                    ", terminationCause=" + terminationCause +
+                    '}';
+        }
+
+        @Override
+        protected void finalize() throws Throwable {
+            if (!completed) {
+                LOGGER.warn("Abandoned ticket detected: {}. Ignoring ticket.", this);
+                ignored();
+            }
+            super.finalize();
+        }
+
+        static void setTerminationCause(Ticket ticket, @Nullable Object result) {
+            if (ENABLE_DEBUGGING && ticket instanceof DebugTrackingDelegatingTicket) {
+                DebugTrackingDelegatingTicket t = (DebugTrackingDelegatingTicket) ticket;
+                t.terminationCause = result;
```

Comment:
One more idea here could be to check the instance of the `result` and if it's a response, we can do `transformMessageBody` to intercept message body (similar to `HttpMessageDiscardWatchdogClientFilter`) and then log if someone subscribed to ir and what was its terminal event (if any).